packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -680,7 +687,15 @@ https://flutter.dev/docs/development/packages-and-plugins/developing-packages#pl
         offline: boolArg('offline'),
         generateSyntheticPackage: false,
       );
-      await project.ensureReadyForPlatformSpecificTooling(checkProjects: pluginExampleApp);
+
+      await project.generateInitialPlatformSpecificTooling(
+        androidPlatform: templateContext['android'] as bool ?? false,
+        iosPlatform: templateContext['ios'] as bool ?? false,
+        linuxPlatform: templateContext['linux'] as bool ?? false,
+        macOSPlatform: templateContext['macos'] as bool ?? false,
+        windowsPlatform: templateContext['windows'] as bool ?? false,
+        webPlatform: templateContext['web'] as bool ?? false,
```

Comment:
These templateContext values are only true if the corresponding feature is on.
https://github.com/flutter/flutter/blob/73301a35a22824edff155836461747d8e04d448d/packages/flutter_tools/lib/src/commands/create.dart#L423-L426